src/app/content/highlights/components/Card.tsx

```diff
@@ -101,25 +101,28 @@ const Card = (props: CardProps) => {
   const location = React.useMemo(() => {
     return props.page && getHighlightLocationFilterForPage(locationFilters, props.page);
   }, [locationFilters, props.page]);
+  const locationFilterId = location && stripIdVersion(location.id);
+  const { page, book } = props;
 
-  const isMobile = useMatchMobileMediumQuery();
+  const onRemove = React.useCallback(
+    () => {
+      if (props.data && page && locationFilterId) {
```

Comment:
Is this here just to fix types? I'm wondering if it would cause the remove highlight button to not work at all if something is wrong with page or locationFilterId. Would that matter?

Reply:
I did not change the code of onRemove; I just put it in useCallback. The call to props.remove requires `page.id` and `locationFilterId`. I don't have a real understanding of how that works.

Reply:
Looking at it again, if page or locationFilterId are null or undefined you don't render anything... you could probably move the `onRemove` definition to below the if (...) { return null; } and not have to have this condition here but it's fine.

Reply:
Hooks have to be called every time their caller is called. It's part of how React[ keeps track of them](https://legacy.reactjs.org/docs/hooks-rules.html#explanation). The way around it would be to put the part after the `return null` into a separate function (which would include the onRemove), which seemed like too much trouble to me.

Reply:
On second thought, I went ahead and broke that down.